legend-engine-pure-code-compiled-core/src/main/resources/core/pure/binding/externalFormat/externalFormatContract.pure

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::external::shared::format::*;
+import meta::external::shared::format::binding::*;
+import meta::external::shared::format::binding::toPure::*;
+import meta::external::shared::format::binding::fromPure::*;
+import meta::external::shared::format::binding::validation::*;
+import meta::external::shared::format::executionPlan::*;
+import meta::external::shared::format::metamodel::*;
+
+import meta::pure::generation::metamodel::*;
+
+import meta::pure::metamodel::path::*;
+
+Class meta::external::shared::format::ExternalFormatContract<T>
+[
+  externalFormatMetamodelType : checkSuperType($this.externalFormatMetamodel, SchemaDetail)
+]
+{
+   id                               : String[1];   
+   contentTypes                     : String[*];
+   
+   externalFormatMetamodel          : Class<T>[1];
+
+   externalFormatToPureDescriptor   : ExternalFormatToPureDescriptor<SchemaToModelConfiguration>[0..1];
+   externalFormatFromPureDescriptor : ExternalFormatFromPureDescriptor<ModelToSchemaConfiguration>[0..1];
+   externalFormatBindingValidator   : Function<{Binding[1] -> BindingDetail[1]}>[0..1];
+
+   externalFormatPlanDescriptor     : ExternalFormatPlanDescriptor[0..1];
+}
+
+Class meta::external::shared::format::ExternalFormatToPureDescriptor<U>
+[
+  configurationType : checkSuperType($this.configuration, SchemaToModelConfiguration)
+]
+{
+  configuration        : Class<U>[1];
+  generator            : Function<{SchemaSet[1], U[1] -> Binding[1]}>[1];
+  generatorWithDebug   : Function<{SchemaSet[1], U[1] -> Binding[1]}>[0..1];
+  defaultConfig        : U[1];
+}
+
+Class meta::external::shared::format::ExternalFormatFromPureDescriptor<V>
+[
+  configurationType : checkSuperType($this.configuration, ModelToSchemaConfiguration)
+]
+{
+  configuration        : Class<V>[1];
+  generator            : Function<{V[1] -> Binding[1]}>[1];

```

Comment:
Should we return SchemaSet from this generator?
This would be beneficial for SchemaGeneration and seems more intuitive for a generator of SchemaToModelGenerator